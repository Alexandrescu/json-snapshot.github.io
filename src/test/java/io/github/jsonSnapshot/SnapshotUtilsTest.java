package io.github.jsonSnapshot;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

import static io.github.jsonSnapshot.SnapshotMatcher.expect;
import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;

@RunWith(MockitoJUnitRunner.class)
public class SnapshotUtilsTest {

    @Mock
    private FakeObject fakeObject;

    @BeforeClass
    public static void beforeAll() {
        start();
    }

    @AfterClass
    public static void afterAll() {
        validateSnapshots();
    }


    @Test
    public void shouldExtractArgsFromMethod() {

        fakeObject.fakeMethod("test1", 1L, Arrays.asList("listTest1"));
        fakeObject.fakeMethod("test2", 2L, Arrays.asList("listTest1", "listTest2"));

        HashMap<?, ?> fakeMethodArgs = SnapshotUtils.extractArgs(fakeObject, "fakeMethod", String.class, Long.class, List.class);
        expect(fakeMethodArgs)
                .toMatchSnapshot();

    }

}
