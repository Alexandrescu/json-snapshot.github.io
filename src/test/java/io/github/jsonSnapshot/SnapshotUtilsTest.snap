io.github.jsonSnapshot.SnapshotUtilsTest.shouldExtractArgsFromMethod=[
  {
    "FakeObject.fakeMethod": [
      {
        "arg0": "test1",
        "arg1": 1,
        "arg2": [
          "listTest1"
        ]
      },
      {
        "arg0": "test2",
        "arg1": 2,
        "arg2": [
          "listTest1",
          "listTest2"
        ]
      }
    ]
  }
]